Automattic/jetpack
# My Jetpack: Repeat Jetpack AI features on all tiers

Fixes #34468.

## Proposed changes:

* Repeat the list of features on all tiers
* Highlight the number of requests using bold style

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Run this branch on your testing site and point the site to your sandbox
* Use the filters we have in place to enable/disable the tiers and also to change which tier your testing site already have:

```
// set to true or false to enable/disable the tiers
add_filter( 'jetpack_ai_tier_plans_enabled', '__return_true' );

add_filter( 'jetpack_ai_current_period_requests_count', function() { return 0; } );
add_filter( 'jetpack_ai_all_time_requests_count', function() { return 0; } );

// set to 0, 100, 200, 500, 750, 1000 to change the tier the site is currently on
add_filter( 'jetpack_ai_tier_licensed_quantity', function() { return 0; } );
```

* On all the test scenarios, reload the Jetpack AI interstitial to confirm it's looking as it should 
   * To open it, you can go to `Jetpack > My Jetpack` and then scroll down to the Jetpack AI card, then click the button there
* With tiers disabled and the site on the free plan (`jetpack_ai_tier_licensed_quantity` return 0), you should see the old interstitial, with no changes:

<img width="500" alt="Screenshot 2023-12-08 at 09 51 29" src="https://github.com/Automattic/jetpack/assets/6760046/c346cb39-b3bf-4065-aa7c-15f8d0290204">

* With tiers enabled and the site on the free plan (`jetpack_ai_tier_licensed_quantity` return 0), you should see the new interstitial, with all the features, and the highlight on the requests count:

| Before | After |
| --- | --- |
| <img width="500" alt="Screenshot 2023-12-08 at 10 31 13" src="https://github.com/Automattic/jetpack/assets/6760046/07b27813-9153-4c17-a9e4-2d92219f2e62"> | <img width="500" alt="Screenshot 2023-12-08 at 09 52 01" src="https://github.com/Automattic/jetpack/assets/6760046/c64fd644-6869-4ccf-abee-ba657f3b112c"> |

* With tiers enabled and the site on the 100 requests plan (`jetpack_ai_tier_licensed_quantity` return 100), you should see the new interstitial, updated with all the features, and the highlight on the requests count:

| Before | After |
| --- | --- |
| <img width="500" alt="Screenshot 2023-12-08 at 10 32 00" src="https://github.com/Automattic/jetpack/assets/6760046/01239301-46cd-4b35-9ae9-bb3729f832c3"> | <img width="500" alt="Screenshot 2023-12-08 at 09 52 49" src="https://github.com/Automattic/jetpack/assets/6760046/76f7075e-ce25-40eb-b81a-56ec443077aa"> |

* The same is true for all the other tiers (example, the 500 requests):

| Before | After |
| --- | --- |
| <img width="500" alt="Screenshot 2023-12-08 at 10 35 59" src="https://github.com/Automattic/jetpack/assets/6760046/ba2e1eab-3be7-4140-959f-f2996c79ea65"> | <img width="500" alt="Screenshot 2023-12-08 at 10 02 45" src="https://github.com/Automattic/jetpack/assets/6760046/830dc6f2-4b30-425d-b2b0-016d78a560da"> |

* When on the 1000 requests tier, you should see the same result as before, the Contact Us call to action:

<img width="500" alt="Screenshot 2023-12-08 at 10 04 33" src="https://github.com/Automattic/jetpack/assets/6760046/72bdceba-ce99-47c2-8c2a-1cfd11521fe7">
